internal/opamp/api.go

```
@@ -232,15 +247,21 @@ func (s *opampServer) OnMessage(conn opamp.Connection, message *protobufs.AgentT
 // Typically, preceded by OnDisconnect() unless the client misbehaves or the
 // connection is lost.
 func (s *opampServer) OnConnectionClose(conn opamp.Connection) {
-	ctx, span := tracer.Start(context.TODO(), "opamp/disconnected")
+	ctx, span := tracer.Start(context.TODO(), "opamp/OnConnectionClose")
 	defer span.End()
 
-	agentID := s.connections.agentID(conn)
+	state, _ := s.connections.OnConnectionClose(conn)
```

Are we ignoring an error here?

---

The second return value is the number of connected agents